src/SystemHandle.cpp

```diff
@@ -69,22 +69,12 @@ class SystemHandle : public virtual FullSystem
          */
         try
         {
-            if (configuration["participant"])
-            {
-                participant_ = std::make_unique<Participant>(configuration["participant"]);
-            }
-            else
-            {
-                logger_ << utils::Logger::Level::WARN
-                        << "Participant not provided in configuration file. "
-                        << "A participant using the default transport locators "
-                        << "and Domain ID 0 will be created." << std::endl;
-
-                participant_ = std::make_unique<Participant>();
-            }
+            // TODO Add a warning that was here in Participant build_participant creation
+            participant_ = std::make_unique<Participant>(configuration);
```

Comment:
Not sure if I am convinced with this.
In my opinion:
* We should create a CMake flag `IS_FASTDDS_SH_DATABROKER` or similar.
* The flag should be *disabled* by default.
* cpp files should be included in the library conditioned to that flag. This way, we avoid as much as possible to fill the code with `#ifdef IS_FASTDDS_SH_DATABROKER` clauses

Reply:
What do you think of this @laura-eprosima ?

Reply:
Finally the part of the code regarding the databroker is minimum, just method `get_databroker_qos` and an if clause in `build_participant`.